core/trino-spi/src/main/java/io/trino/spi/connector/JoinApplicationResult.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public final class JoinApplicationResult<T>
+{
+    private final T tableHandle;
+    private final Map<ColumnHandle, ColumnHandle> columnHandles;
```

Comment:
I think we need to remap them separately, so

`newLeftColumnHandles`, `newRightColumnHandles`

Reply:
I can do that but I do not really see a reason why. Can you elaborate?

Reply:
the equality between column handles is often based on the column attributes (like name, see `JdbcColumnHandle`)
so we cannot discern between eg `name` column coming from left table and `name` coming from right

Reply:
Makes sense. Changed.